content/docs/engine-dev/tools/project-manager/source-code-map.md

```diff
@@ -0,0 +1,110 @@
+
+{{< note >}}
+This information is for developers of the **Project Manager** tool. If you're a user working with Project Manager, please refer to the [Project Manager User Guide](/docs/user-guide/project-config/project-manager).
+{{< /note >}}
+
+{{< note >}}
+This contains brief summaries of various class's functionality and role. If available, there are additional links in the table of contents, either with expanded information, or github links.
+{{< /note >}}
+
+## Map
+- <b>Startup Classes</b>
+    - main.cpp [[cpp file](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/main.cpp) ]
+        - Entry point for the Project Manager. Initiates the [Application main loop](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/main.cpp#L28).
+    - [Application](#application)  [[header](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/Application.h) ] [[cpp file](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/Application.cpp) ]
+        - Houses core startup code and configuration loading for Qt framework, Logging, Python Bindngs and O3DE Engine Registration. Facilitates prior housework to starting up the application main loop, and clean up when tearing down.
+    - [ProjectManagerWindow](#projectmanagerwindow)  [[header](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/ProjectManagerWindow.h) ] [[cpp file](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/ProjectManagerWindow.cpp) ]
+        - Constructs the application's `QMainWindow` and `DownloadController`, and prepares all relevant transition screens for the Project Manager.
+    - [ScreensCtrl](#screensctrl)  [[header](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/ScreensCtrl.h) ] [[cpp file](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/ScreensCtrl.cpp) ]
+        - Container structure for all Project Manager Screens and GUI Widgets. Also houses primary transition code for facilitating Project Manager's business logic.
+    - ScreenFactory  [[header](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/ScreenFactory.h) ] [[cpp file](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/ScreenFactory.cpp) ]
+        - Helper class for ScreensCtrl that routes `ProjectManagerScreen` enums to appropriate ScreenWidget constructor, invokes that construtor, and returns an instance of a given screen for the Project Manager's business logic.
+
+
+### Application
+
+Inherits from `AzFramework::Application`.
+
+The functions of the Application class are exclusively run by `main.cpp`.
+
+The two primary functions to keep track of are `Init` and `Run`. [`TearDown`](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/Application.cpp#L260-L272) is invoked at `Application`'s deconstruction.
```

Comment:
I agree that it's super useful to have links to specific lines and blocks of source code, however at the same time I worry about the confusion it could lead to when the source code inevitably changes and the links no longer line up with the intended line or range. I don't have a good solution at the moment, but I wanted to throw this out there. It is possible to create permalinks to specific commits that will always look correct, but then if the code changes, the viewer won't see the new code.

Reply:
I've thought about this as well. One potential idea is to embrace the idea that the documentation was written at a certain commit in time, and update all links to match up with that particular commit. This would require saying up at the top what commit the documentation is calibrated for.

This can guarantee accuracy, but anytime a commit significantly changes part of the code, we would have to come back here and recalibrate for the latest commit.

Even with this disadvantage, I think it's better than nothing, because at least it can provide some development context for where the code was at in an earlier point, which can still make it easier to navigate the present code.

Reply:
Based on the commit frequency of the project manager, I believe we can choose to prioritize what commits we update the documentation with. We tend to have more work done closer to stabilization and releasing of a new version of O3DE. As such, major changes at that time make sense to update documentation with.

Another potential idea for the documentation to help combat this issue in the long term would be to provide guidelines/advice on how to navigate the github codebase for new contributors. It's basically a reality that we need to compare and consider diffs, and build up a virtual timeline of what changes were made and why. This can help teach newcomers to identify discrepancies already present in the codebase as well.

Reply:
sounds good, I will update all links to point to these permalinks (which should point to the relevant commit at the time of doc writing)